xarray/coding/times.py

```diff
@@ -197,9 +197,12 @@ def _decode_cf_datetime_dtype(data, units, calendar, use_cftime):
 def _decode_datetime_with_cftime(num_dates, units, calendar):
     if cftime is None:
         raise ModuleNotFoundError("No module named 'cftime'")
-    return np.asarray(
-        cftime.num2date(num_dates, units, calendar, only_use_cftime_datetimes=True)
-    )
+    if num_dates.size > 0:
+        return np.asarray(
+            cftime.num2date(num_dates, units, calendar, only_use_cftime_datetimes=True)
+        )
+    else:
+        raise ValueError("Can't decode 0-sized times with cftime.")
```

Comment:
@spencerkclark do you have an idea for a solution here? `np.zeros(shape=0, dtype=object)`

Reply:
Indeed I think that's what cftime *should* produce here.  For instance if the empty array is of dtype `float64` that's what it produces:
```
>>> cftime.num2date(np.array([], dtype="float64"), "days since 1970-01-01", calendar="proleptic_gregorian", only_use_cftime_datetimes=True)
array([], dtype=object)
```
But if the empty array is of dtype `int64` we get the error:
```
>>> cftime.num2date(np.array([], dtype="int64"), "days since 1970-01-01", calendar="proleptic_gregorian", only_use_cftime_datetimes=True)Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/cftime/_cftime.pyx", line 604, in cftime._cftime.num2date
  File "src/cftime/_cftime.pyx", line 442, in cftime._cftime.scale_times
  File "<__array_function__ internals>", line 180, in amin
  File "/Users/spencer/Software/miniconda3/envs/xarray-tests-py38/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 2916, in amin
    return _wrapreduction(a, np.minimum, 'min', axis, None, out,
  File "/Users/spencer/Software/miniconda3/envs/xarray-tests-py38/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 86, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
ValueError: zero-size array to reduction operation minimum which has no identity
```
I'll open an issue in the cftime repo.  In the meantime I might spell it just as `np.array([], dtype=object)` rather than go through the extra hoop of `np.zeros`.